### ISSUE TITLE
No imports, is this even maintaned?

### ISSUE THREAD

**Issue by andreasdc** [eclipse-jdt/eclipse.jdt.ui#434]
Type
Map<String, List<Object>> map = Maps.newConcurrentMap();

You can't import any of these classes.

**Comment by jjohnstn** [comment]
Hello @andreasdc.  I tried your example with 4.27 pre-release and it works fine except that I don't have Google Guava Maps in my classpath so doing CTRL+Shift+O adds imports for Map and List, but not Maps.  It flags List for not having a type parameter.

What version of Eclipse are you using?  By any chance do you have a Preferences->Type Filter that is getting in the way?

**Comment by andreasdc** [comment]
Version: 2022-12 (4.26.0)
Build id: 20221201-1913
![image](https://user-images.githubusercontent.com/82218780/222827866-885e5f9d-9c6f-41ac-b7e9-f08ca9db7a95.png)

**Comment by JanecekPetr** [comment]
Hello!
This is still very unclear. There are literally millions of developers using Eclipse for Java development daily, so do not worry, it's (mostly) not broken :).

Clicking on an unimported class indeed does not import it. If you _hover_ over the class, you will get a popup with options to import the relevant classes. You can also invoke the Quick Fix action on the class (Ctrl+1 by default) or even do a Content Assist (Ctrl+Space by default) or invoke Optimize Imports (Ctrl+Shift+O, but it can also be set up to run automatically after a file save via Save Actions).

If this still does not work for you, please show us your project, something is misconfigured somewhere.

**Comment by andreasdc** [comment]
Try doing private Map<String, List> map = Maps.newConcurrentMap();

**Comment by JanecekPetr** [comment]
> Try doing `private Map<String, List> map = Maps.newConcurrentMap();`

I have, of course. In a proper Java project with Guava on the classpath, it immediately imports the correct classes for me when doing any of the above mentioned actions. When I had two different `Maps` classes on the classpath, Organize imports only imported `Map` for me and I had to invoke a Quick Fix (or hover over) the `Maps` class.

In short, it works as it should. If it does not work for you, please show us your project, or send us a whole example project showcasing the problem, something is wrongly configured. Can you run a simple Hello world from your project to make sure it builds?

**Comment by andreasdc** [comment]
It works weird, when I posted the issue it wasn't working, when I posted the comment it wasn't working, right now it's working. I will try with these shortcuts too.

**Comment by andreasdc** [comment]
Still not fixed.

**Comment by JanecekPetr** [comment]
> Still not fixed.

Hello! I'm not an eclipse engineer, I'm a random lurker.
That said, as a fellow engineer you should appreciate that it's hard to fix a problem without proper diagnosis. If this was a widespread problem, many other people would be reporting this as well. They don't, so this is somehow isolated to your environment. Let's find out what stands out!

What version of eclipse are you using? Have you tried updating to 2023-03 which came out in the meantime? Is there anything notable about your setup, like some plugins, your OS, anything else?

What exactly are you trying and what is happening instead? Do you see anything in the Error log view after the import action fails (Window > Show view > Other, then General > Error Log)? Last time you we saying you're clicking on the classes you're trying to import, and I tried to explain that's not how to import classes which is why I'm asking for the exact sequence of actions.

Help us reproduce this please.

**Comment by andreasdc** [comment]
> Try doing `private Map<String, List> map = Maps.newConcurrentMap();`

This

**Comment by trancexpress** [comment]
> > Try doing `private Map<String, List> map = Maps.newConcurrentMap();`
> 
> This. For example you have new HashMap and you want to change non imported maps to this, it will show that you dont have it referenced libraries, you will need to put the maps inside a method for example.

You'll need to provide more steps, maybe a screen recording will help understand what your problem is.

E.g. this works as expected from my POV, so provide a video showcasing that it doesn't:

https://user-images.githubusercontent.com/24752155/229618096-938f2581-5b6e-4ddc-b8cf-2fc156d9af7d.mp4

**Comment by andreasdc** [comment]
Have imported Map and List then try to change maps like I described above.

**Comment by trancexpress** [comment]
OK, I see what you mean:

```
import java.util.List;
import java.util.Map;

public class Snippet {
	private Map<String, List> map = new HashMap<>();

	private void m(HashMap<String, List> t) {
		Map<String, List> map = new HashMap<>();
	}
}
```

https://user-images.githubusercontent.com/24752155/229629523-be5d6d74-6113-4049-8010-1182e53cbf03.mp4

**Comment by andreasdc** [comment]
Yes, I think that's it. :)

**Comment by trancexpress** [comment]
First commit I see this with is: 546a6978432fa5f2b01083919364848dd89805dc

@jjohnstn can you take a look here?

**Comment by jjohnstn** [comment]
Yes, I see the problem.

**Comment by trancexpress** [comment]
Thanks @jjohnstn !

**Comment by andreasdc** [comment]
It is not fixed in the latest version that I have, is it not released yet?

**Comment by JanecekPetr** [comment]
> is it not released yet

Indeed. You can see the Milestone marked as 4.28 M1 which means this targets 4.28, that's another name for the 2023-06 release (see https://www.eclipse.org/eclipse/news/4.28/).

**Comment by andreasdc** [comment]
How long until the release?

**Comment by jjohnstn** [comment]
The full 4.28 release won't be until June 14.  You can either install a new 4.28 M1 EPP package or SDK that matches what you are currently using to try it out without affecting your current set-up or else you can update your current Eclipse.  To get a new 4.28 M1 Eclipse package you can go to:

https://www.eclipse.org/downloads/packages/release/2023-06/m1

A 4.28 Eclipse SDK can be downloaded from:

https://download.eclipse.org/eclipse/downloads/drops4/S-4.28M1-202304051800/

If you want to update your current Eclipse to 4.28 M1, you can specify the repo at:

https://download.eclipse.org/eclipse/updates/4.28-I-builds/I20230405-1800/

and do an update.

**Comment by andreasdc** [comment]
I did the update and now I can't do anything in Eclipse:
An internal error occurred during: "Requesting Java AST from selection".
'boolean com.ibm.icu.text.UTF16.isSurrogate(char)'

**Comment by andreasdc** [comment]
Also
`Cannot complete the install because of a conflicting dependency.
  Software being installed: Eclipse Platform 4.28.0.v20230405-1811 (org.eclipse.platform.feature.group 4.28.0.v20230405-1811)
  Software currently installed: Eclipse IDE for Java Developers 4.27.0.20230309-1200 (epp.package.java 4.27.0.20230309-1200)
  Only one of the following can be installed at once: 
    Compare Support for Word 1.2.100.v20180515-0858 (org.eclipse.compare.win32 1.2.100.v20180515-0858)
    Compare Support for Word 1.3.0.v20230317-0901 (org.eclipse.compare.win32 1.3.0.v20230317-0901)
    Compare Support for Word 1.0.200.v20130514-1224 (org.eclipse.compare.win32 1.0.200.v20130514-1224)
    Compare Support for Word 1.2.900.v20220812-1406 (org.eclipse.compare.win32 1.2.900.v20220812-1406)
    Compare Support for Word 1.2.800.v20200127-1343 (org.eclipse.compare.win32 1.2.800.v20200127-1343)
  Cannot satisfy dependency:
    From: Eclipse IDE for Java Developers 4.27.0.20230309-1200 (epp.package.java 4.27.0.20230309-1200)
    To: org.eclipse.equinox.p2.iu; org.eclipse.platform.feature.group [4.27.0.v20230302-0300,4.27.0.v20230302-0300]
  Cannot satisfy dependency:
    From: Eclipse Platform 4.27.0.v20230302-0300 (org.eclipse.platform.feature.group 4.27.0.v20230302-0300)
    To: org.eclipse.equinox.p2.iu; org.eclipse.compare.win32 [1.2.900.v20220812-1406,1.2.900.v20220812-1406]
  Cannot satisfy dependency:
    From: Eclipse Platform 4.28.0.v20230405-1811 (org.eclipse.platform.feature.group 4.28.0.v20230405-1811)
    To: org.eclipse.equinox.p2.iu; org.eclipse.compare.win32 [1.3.0.v20230317-0901,1.3.0.v20230317-0901]`

**Comment by andreasdc** [comment]
Also after adding the rep
![image](https://user-images.githubusercontent.com/82218780/233532186-6c69b863-746b-454b-83ad-1187685d1107.png)

**Comment by jjohnstn** [comment]
Something that you have installed in your Eclipse set-up still requires an older version of icu4j as the method in question was removed in newer versions.  I did see an old StackOverFlow issue that reported an error with that icu4j method and suggested working around it by disabling Mark Occurrences: https://stackoverflow.com/questions/74780857/boolean-com-ibm-icu-text-utf16-issurrogatechar  Not sure if that would help you.

**Comment by andreasdc** [comment]
Still active.
